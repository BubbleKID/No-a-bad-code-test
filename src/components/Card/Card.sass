//variables
$border-color: #d1d1d1
$font-color: #424242
$border-color-hover: #00ac88
$arrow-color: #00ac88
$file-color: #d1d1d1
$file-color-hover: #424242
$text-underline-color: #ff69b4
$featured-arrow-color: #fff
$featured-arrow-bg-color: rgba(0, 0, 0, 0.4)
$featured-font-color: #fff
$featured-bg-0: #00a4c7
$featured-bg-25: #00ac88
$featured-bg-50: #1e539e
$featured-bg-75: #00ac88
$featured-bg-100: #00a4c7

.card 
    font-family: Open Sans
    font-weight: 700
    font-size: 16px
    padding: 50px 20px
    border: 1px solid $border-color
    color: $font-color
    margin: 20px
    position: relative
    transition: 0.5s all
    &:hover
        border-color: $border-color-hover
    &__title
        cursor: pointer
        a
            background-color: hotpink
            background: linear-gradient(to right, transparent, transparent), linear-gradient(to right, $text-underline-color, $text-underline-color)
            background-size: 100% 0.1em, 0 0.1em
            background-position: 100% 100%, 0 100%
            background-repeat: no-repeat
            transition: background-size 800ms
            &:hover, &:focus 
                background-size: 0 0.1em, 100% 0.1em
    &__arrow-container
        cursor: pointer
        position: absolute
        right: 0
        bottom: 0
    &__arrow
        cursor: pointer
        padding: 15px
        width: 24px
        transition: all .2s ease-in-out
        path
            fill: $arrow-color
        &:hover
            transform: scale(1.2)
    &__document-btn
        display: flex
        align-items: center
        width: 100px
        cursor: pointer
        position: absolute
        left: 25px
        bottom: 25px
        &:hover
            span
                color: $font-color
            .card__document-icon
                path
                    fill: $font-color
        span
            font-weight: 400
            color: $file-color
            font-size: 12px
    &__document-icon
        margin-right: 5px
        width: 20px
        path
            fill: $file-color
    &--featured
        border: none
        background: linear-gradient(-45deg, $featured-bg-0 0%, $featured-bg-25 25%, $featured-bg-50 50%, $featured-bg-75 75%, $featured-bg-100 100%)
        background-size: 400% 400%
        background-repeat: no-repeat
        transition: 1.5s all
        &:hover
            background-position: 100% 100%
        p
            cursor: pointer
            color: $featured-font-color
            font-size: 26px
    &--featured &__title a
        background: none
    &--featured &__description 
        font-size: 14px 
        font-weight: 300
    &--featured &__arrow-container 
        background: $featured-arrow-bg-color
    &--featured &__arrow 
        path
            fill: $featured-arrow-color
