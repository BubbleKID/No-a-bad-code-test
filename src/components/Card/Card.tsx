import './Card.sass';
import React from 'react';
import { ReactComponent as ArrowDownBtn } from '../../images/svg/arrow-down.svg';
import { ReactComponent as ArrowRightBtn } from '../../images/svg/arrow-right.svg';
import { ReactComponent as DocumentIcon } from '../../images/svg/document.svg';

interface Props {
    title?: string,
    featured?: string,
    backroundColor?: string,
    height: number,
    width: number,
    flexGrow?: string,
    link?: string
    description?: string,
    documentSize?: string
}

export default function Card(props: Props) {
    let width: number = props.width;
    props.featured === 'true' ? width = width * 2 + 82 : width;
        
    return (
      <div className={`card ${props.featured === 'true' ? 'featured' : ''}`} style={{width: width, height: props.height}} >
        <p className="card-title"><a>{props.title}</a></p>
        { (props.description && props.featured) && <p className="description">{props.description}</p> }
        { props.documentSize ? <ArrowDownBtn className="arrow"/> : <ArrowRightBtn className="arrow"/> }
        { props.documentSize && 
            <div className="document-btn">
              <DocumentIcon className="document-icon"/>
              <span>PDF({props.documentSize})</span>
            </div> 
        }
      </div>
    );
}