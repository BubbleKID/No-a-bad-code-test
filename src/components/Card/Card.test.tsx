import React from 'react';
import TestRenderer  from 'react-test-renderer';
import Card from './Card';

test('Card should be rendered', () => {
    const component = TestRenderer.create(<Card height={200} width={200}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Card should be rendered if it has documentSize prop', () => {
    const component = TestRenderer.create(<Card height={200} width={200} documentSize={"100k"}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});