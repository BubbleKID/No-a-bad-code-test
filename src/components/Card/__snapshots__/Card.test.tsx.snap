// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be rendered 1`] = `
<div
  className="card "
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <p
    className="card-title"
  >
    <a />
  </p>
  <span
    className="arrow"
  />
</div>
`;

exports[`Card should be rendered if it has documentSize prop 1`] = `
<div
  className="card "
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <p
    className="card-title"
  >
    <a />
  </p>
  <span
    className="arrow"
  />
  <div
    className="document-btn"
  >
    <span
      className="document-icon"
    />
    <span>
      PDF(
      100k
      )
    </span>
  </div>
</div>
`;
